lib/providers/race_events.dart

```diff
@@ -43,66 +42,16 @@ Stream<Map<String, List<int>>> teamScores(TeamScoresRef ref, String eventId) {
   return controller.stream;
 }
 
-// TODO change to immutable map
-class TeamPositionNotifier extends StateNotifier<Map<String, LatLng>> {
-  final DatabaseReference firebaseDb;
-
-  TeamPositionNotifier(this.firebaseDb, String eventId)
-      : super(<String, LatLng>{}) {
-    final logger = getLogger("TeamPositionNotifier");
-    logger.i("initializing a TeamPositionNotifier provider");
-
-    final dbScoresRef = firebaseDb.child('scores').child(eventId);
-
-    // TODO get the teams from teams document instead of scores document
-    dbScoresRef.onValue.listen((event) {
-      final teamsAndScores = event.snapshot.value as Map<dynamic, dynamic>;
-
-      logger.i(
-          "getting new scores data from firebase: ${teamsAndScores.toString()}");
-
-      for (final team in teamsAndScores.keys.toList()) {
-        final dbRef = firebaseDb
-            .child('traces')
-            .child(eventId)
-            .child(team)
-            .child('lastPosition');
-        dbRef.onValue.listen((DatabaseEvent event) {
-          if (event.snapshot.value == null) {
-            return;
-          }
-          final data = event.snapshot.value as String;
-          logger.i("updating $team position: ${data.toString()}");
-          final latlong = data.split(', ');
-          var latitude = double.parse(latlong[0]);
-          var longitude = double.parse(latlong[1]);
-          var newMap = Map.of(state);
-          newMap[team] = LatLng(latitude, longitude);
-          state = newMap;
-        });
-      }
-    });
-  }
-}
-
-StateNotifierProvider<TeamPositionNotifier, Map<String, LatLng>>
-    teamPositionProvider(String eventId) =>
-        StateNotifierProvider<TeamPositionNotifier, Map<String, LatLng>>((ref) {
-          final db = ref.read(firebaseDbProvider);
-
-          return TeamPositionNotifier(db, eventId);
-        });
-
 @riverpod
 class CurrentRound extends _$CurrentRound {
   @override
-  int build() {
-    // TODO should be synced with db
+  int build(String eventId) {
+    // TODO should get the current round from firebase
```

Comment:
It still isn't real number, yeah?

Reply:
Depends on what you mean by real hahaha When the event starts, the round is 0, then it is increased by admin and the number in this provider will be the actual round number. The problem is that if you leave the admin page or relunch the app this provider would say that currently, it is round 0. I think we should iterate in this constructor through messages and decide that is the current round of the event.